model/Hierarchy.php

```diff
@@ -610,9 +622,13 @@ public function numChildren($cache = true) {
 	 */
 	public function stageChildren($showAll = false) {
 		$baseClass = ClassInfo::baseDataClass($this->owner->class);
+		$hidden = $this->owner->config()->hide_from_hierarchy;
 		$staged = $baseClass::get()
 			->filter('ParentID', (int)$this->owner->ID)
 			->exclude('ID', (int)$this->owner->ID);
+		if (count($hidden) > 0) {
```

Comment:
`!empty()` is slightly better here as we aren't interested in how many there are, just that there are some